front/components/assistant_builder/AssistantBuilder.tsx

```diff
@@ -235,15 +246,19 @@ export default function AssistantBuilder({
   dustApps,
   initialBuilderState,
   agentConfigurationId,
+  flow,
 }: AssistantBuilderProps) {
   const router = useRouter();
   const sendNotification = React.useContext(SendNotificationsContext);
   const slackDataSource = dataSources.find(
     (ds) => ds.connectorProvider === "slack"
   );
+  const initialScope =
+    flow === "workspace_assistants" ? "workspace" : "private";
 
   const [builderState, setBuilderState] = useState<AssistantBuilderState>({
     ...DEFAULT_ASSISTANT_STATE,
+    scope: initialBuilderState?.scope ?? initialScope,
```

Comment:
Will initialScope/defaultScope ever be taken into account since initialBuilderState.scope defaults to DEFAULT_ASSISTANT_STATE which has scope set to `private` ?

Reply:
initialBuilderState can be null (when an assistant is created)